player.js

```diff
@@ -0,0 +1,124 @@
+// remove startingPoint and nextSpace
+function Player(name) {
+    this.name = name;
+    this.marker = null;
+
+    this.point = null; // should be a Point object
+    this.nextSpace = null; // [x, y] // depends on the angular Space.x, Space.y
+
+    // in each Space.points array, find this specific point and get the position (integer) inside this space.
+    this.nextSpacePointsIndex = null;
+
+    // maximun 3 tiles
+    this.tiles = [];
+
+    // if a player dies, it will be changed to false
+    this.canPlay = true;
+}
+
+Player.prototype.pickMarker = function (marker) {
+    this.marker = marker;
+}
+
+// Placing marker for the first time
+Player.prototype.placeMarker = function (point, nextSpace) {
+    this.point = point;
+    this.point.travelled = true;
+    this.nextSpace = nextSpace;
+    // in each Space.points array, find this specific point and get the position (integer) inside this space.
+    this.nextSpacePointsIndex = this.nextSpace.points.indexOf(this.point)
+}
+
+function rotateTileCw(tile) {
+    var addTwo = tile.map(function (connection) {
+        return connection + 2;
+    });
+    addTwo.unshift(addTwo.pop());
+    addTwo.unshift(addTwo.pop());
+    return Player.placeTile(addTwo);
+}
+
+function rotateTileCcw(tile) {
+    var minusTwo = tile.map(function (connection) {
+        return connection - 2;
+    });
+    minusTwo.push(minusTwo.shift());
+    minusTwo.push(minusTwo.shift());
+    return Player.placeTile(minusTwo);
+}
+
+// when we get the tile, it is already rotated or not, from angular.
+Player.prototype.placeTile = function (tile) {
+    var index = this.tiles.indexOf(tile)
+    this.tiles.splice(index, 1);
+
+    for (var i = 0; i < tile.length; i++) {
+        this.nextSpace.points[i].neighbors.push(points[tile[i]]);
+    }
+
+    let playersWhoCanPlay = Game.players.filter(function (player) {
```

Comment:
Player should not be have to generate the  playersWhoCanPlay array. This should be handled by the Game object